aptos-move/framework/aptos-token/sources/token.move

```diff
@@ -775,64 +765,35 @@ module aptos_token::token_v1 {
         assert!(token_data_id.creator == signer::address_of(account), ENO_MINT_CAPABILITY);
         let creator_addr = token_data_id.creator;
         let all_token_data = &mut borrow_global_mut<Collections>(creator_addr).token_data;
+        assert!(table::contains(all_token_data, token_data_id), ETOKEN_NOT_PUBLISHED);
         let token_data = table::borrow_mut(all_token_data, token_data_id);
 
         assert!(token_data.supply + amount <= token_data.maximum, 1);
 
         token_data.supply = token_data.supply + amount;
 
-        // we add more tokens with serial_number 0
+        // we add more tokens with property_version 0
         let token_id = create_token_id(token_data_id, 0);
         deposit_token(account,
             Token{
                 id: token_id,
-                value: amount
+                amount
+            }
+        );
+        event::emit_event<MintTokenEvent>(
+            &mut borrow_global_mut<Collections>(creator_addr).mint_token_events,
+            MintTokenEvent {
+                id: token_data_id,
+                amount,
             }
         );
-        token_id
-    }
-
-    public fun create_token_id(token_data_id: TokenDataId, serial_number: u64): TokenId {
-        TokenId{
-            token_data_id,
-            serial_number,
-        }
-    }
-
-    public fun create_token_data_id(
-        creator: address,
-        collection: String,
-        name: String,
-    ): TokenDataId {
-        TokenDataId { creator, collection, name }
-    }
-
-    public fun create_token_id_raw(
-        creator: address,
-        collection: vector<u8>,
-        name: vector<u8>,
-        serial_number: u64,
-    ): TokenId {
-        TokenId{
-            token_data_id: create_token_data_id(creator, string::utf8(collection), string::utf8(name)),
-            serial_number,
-        }
-    }
 
-    public entry fun burn_script(
-        account: &signer,
-        owner: address,
-        token_id: TokenId,
-        amount: u64,
-    ) acquires TokenStore, Collections {
-        assert!(balance_of(owner, token_id) >= amount, EINSUFFICIENT_BALANCE);
-        let burn_cap = acquire_burn_capability(account, owner, token_id);
-        burn(burn_cap, owner, token_id, amount);
+        token_id
     }
 
-    public fun burn(burn_cap: BurnCapability, owner: address, token_id: TokenId, amount: u64) acquires Collections, TokenStore {
-        assert!(burn_cap.token_id == token_id, ENO_BURN_CAPABILITY);
-        assert!(burn_cap.owner == owner, ENO_BURN_CAPABILITY);
+    public entry fun burn(owner: &signer, token_id: TokenId, amount: u64) acquires Collections, TokenStore {
```

Comment:
We probably should make a decision here:
1) Creator can burn
2) Owner can burn
3) There's a policy at the collection level
4) There's an option in the mutability config

I might suggest we disable this API for now and revisit it.

Reply:
```
3. There's a policy at the collection level
4. There's an option in the mutability config
```
what is the policy and option in your mind?